src/router.rs

```diff
@@ -256,17 +242,4 @@ mod tests {
                 .is_err()
         );
     }
-
-    #[test]
-    fn router_produces_json_not_found() {
-        let route_builder = RouteBuilder::new();
-        let mut req = MockRequest::new(::conduit::Method::GET, "/");
-
-        let (parts, body) = R404(route_builder).call(&mut req).unwrap().into_parts();
-        assert_eq!(parts.status, StatusCode::NOT_FOUND);
-        assert!(matches!(
-            body,
-            Body::Owned(vec) if vec == br#"{"errors":[{"detail":"Not Found"}]}"#
-        ));
-    }
```

Comment:
would it make sense to convert that to a more high-level API test?

Reply:
This particular test moved to the middleware. We could do something under `tests/all.rs` as well. The main difference would be that it would load other middleware layers as well.